### ISSUE TITLE
testbench: hang of mmnormalize_regex_disabled.sh

### ISSUE THREAD

**Issue by rgerhards** [rsyslog/rsyslog#2963]
```
===============================================================================
[mmnormalize_regex_disabled.sh]: test for mmnormalize regex field_type with allow_regex disabled
------------------------------------------------------------
Test: ./mmnormalize_regex_disabled.sh
------------------------------------------------------------
testbench: TZ env var not set, setting it to UTC
imdiag running on port 51613
config rsyslog_mmnormalize_regex_disabled.sh_51613_.conf is:
     1	module(load="../plugins/imdiag/.libs/imdiag")
     2	global(inputs.timeout.shutdown="10000")
     3	$IMDiagServerRun 51613
     4	
     5	:syslogtag, contains, "rsyslogd"  ./rsyslog_mmnormalize_regex_disabled.sh_51613.started
     6	###### end of testbench instrumentation part, test conf follows:
     7	
     8	template(name="hosts_and_ports" type="string" string="host and port list: %$!hps%\n")
     9	
    10	template(name="paths" type="string" string="%$!fragments% %$!user%\n")
    11	template(name="numbers" type="string" string="nos: %$!some_nos%\n")
    12	
    13	module(load="../plugins/mmnormalize/.libs/mmnormalize" allowRegex="off")
    14	module(load="../plugins/imptcp/.libs/imptcp")
    15	input(type="imptcp" port="51613")
    16	
    17	action(type="mmnormalize" rulebase=`echo $srcdir/testsuites/mmnormalize_regex.rulebase`)
    18	action(type="omfile" file=`echo $RSYSLOG_OUT_LOG` template="hosts_and_ports")
rsyslogd: NOTE: RSYSLOG_DEBUG_TIMEOUTS_TO_STDERR activated
rsyslogd: liblognorm error: rulebase file ./testsuites/mmnormalize_regex.rulebase[1]: invalid field type 'regex' [v8.38.0.master try http://www.rsyslog.com/e/2427 ]
RSYSLOG_DYNNAME: rsyslog_mmnormalize_regex_disabled.sh_51613
rsyslogd: Could not create tcp listener, ignoring port 51613 bind-address (null). [v8.38.0.master try http://www.rsyslog.com/e/2077 ]
rsyslogd started, start msg not yet seen, pid  127578
rsyslogd startup msg seen, pid  127578
ports: 51613
starting run 1
Sending file './testsuites/regex_input' 1 times.

00000001 kb sent
runtime: 0.000
End of tcpflood Run
doing shutdown
Shutting down instance 1

command timed out: 1200 seconds without output running ['make', 'check', 'V=0'], attempting to kill
```

**Comment by rgerhards** [comment]
The root cause seems to be here:
```
rsyslogd: Could not create tcp listener, ignoring port 51613 bind-address (null)
```
The (somewhat dirty) trick to get dynamic ports seems to fail occasionally. While I already work on fixing this, it would probably be good to also add the OS error state to above error message. This could shed some light on why it fails. From the test logs, it did not look like the port was actually in use.

**Comment by rgerhards** [comment]
ref: reliable dynamic port assignment is being introduced, a first PR is this: https://github.com/rsyslog/rsyslog/pull/2964

**Comment by rgerhards** [comment]
Apologies for the very late follow-up. rsyslog does not close issues just because they are old, but I am closing this one because the specific failure mode in the retained evidence appears to have been fixed in the current testbench.

The failure recorded here was a bind failure on fixed port `51613`. Current `tests/mmnormalize_regex_disabled.sh` no longer uses that fixed port; it configures `imptcp` with `port="0"` and `listenPortFileName`, so the testbench waits for the dynamically assigned port before starting `tcpflood`.

I validated current `main` in a fresh worktree with:

- `make -j60 check TESTS="mmnormalize_regex_disabled.sh"` - passed, 1/1
- five direct repeats of `./tests/mmnormalize_regex_disabled.sh` under a 60-second timeout - all passed, each using an assigned dynamic TCP port

That makes this stale as a current actionable bug. If this hang reappears on a current rsyslog/testbench checkout, please open a new issue with the fresh test log so we can debug the new failure mode.
